src/App.js

```diff
@@ -0,0 +1,198 @@
+import { React, useState, useEffect } from "react";
+import "./App.css";
+import "bootstrap/dist/css/bootstrap.min.css";
+import { WEATHER_API_KEY } from "./Constants";
+import axios from "axios";
+import {
+  FaLocationArrow,
+  FaTemperatureLow,
+  FaTemperatureHigh,
+  FaCrosshairs,
+} from "react-icons/fa";
+import { getCountryByCode } from "./getCountryByCode";
+import ReactCountryFlag from "react-country-flag";
+import Geocode from "react-geocode";
+
+Geocode.setApiKey(process.env.REACT_APP_LOCATION_API_KEY);
+
+function App() {
+  const [address, setAddress] = useState("");
+  const [locationInfo, setLocationInfo] = useState("");
+  const [countryData, setCountry] = useState("");
+  const [error, setError] = useState(false);
+  const [lat, setLat] = useState(null);
+  const [lng, setLng] = useState(null);
+  const [status, setStatus] = useState(null);
+
+  const getWeather = () => {
+    const weatherApiUrl = `https://api.openweathermap.org/data/2.5/weather?q=${address}&appid=${WEATHER_API_KEY}`;
+    axios
+      .get(weatherApiUrl)
+      .then(function (response) {
+        console.log(response.data);
+        setCountry(getCountryByCode(response.data.sys.country));
+        setLocationInfo(response.data);
+      })
+      .catch(function (error) {
+        setError(true);
+      });
+  };
+
+  const kelvinToFahrenheit = (k) => {
+    return (k - 273.15).toFixed(2);
+  };
+
+  const getLocation = () => {
+    if (!navigator.geolocation) {
+      setStatus("Geolocation is not supported by your browser");
+    } else {
+      setStatus("Locating...");
+      navigator.geolocation.getCurrentPosition(
+        (position) => {
+          setStatus(null);
+          setLat(position.coords.latitude);
+          setLng(position.coords.longitude);
+        },
+        () => {
+          setStatus("Unable to retrieve your location");
+        }
+      );
+    }
+  };
+
+  useEffect(() => {
+    if (lat && lng) {
+      getLocationLatLong(lat, lng);
+    }
+  }, [lat, lng]);
+
+  useEffect(() => {
+    if (address) {
+      getWeather();
+    }
+  }, [address]);
+
+  const getLocationLatLong = (lat, lang) => {
+    Geocode.fromLatLng(lat, lang).then(
+      (response) => {
+        const address = response.results[0].formatted_address;
+        let city, state, country;
+        for (
+          let i = 0;
+          i < response.results[0].address_components.length;
```

Comment:
we should put all of these on one  line... formatting is really bad

Reply:
@Henshall when line length increases from 80 characters then readability gets compromised. To improve readability single line is separated into multiple lines. see industry standard code editor extension https://prettier.io/

Reply:
okay